app/models/person_name.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# A model representing a person's name components in the system.
+# This class is embedded within different types of parent models using a polymorphic
+# association, allowing any model to have name information without duplicating code.
+#
+# @example Embedding in a IndividualMarket::Applicant model
+#   class IndividualMarket::Applicant
+#     include Mongoid::Document
+#     include Mongoid::Timestamps
+#
+#     embeds_one :person_name, class_name: 'PersonName', as: :person_nameable
+#   end
+#
+# @example Embedding in a Person model
+#   class Person
+#     include Mongoid::Document
+#     include Mongoid::Timestamps
+#
+#     embeds_many :names, class_name: 'PersonName', as: :person_nameable
+#   end
+#
+# @note The polymorphic association is used here because multiple models
+#   (like Person, Applicant, Employee, etc.) might need to store name information.
+#   The polymorphic approach allows any model to embed this document without
+#   having to create separate name classes for each parent model type.
+class PersonName
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  # Defines the polymorphic relationship with the parent document
+  # This allows PersonName to be embedded in any model that sets itself as :person_nameable
+  embedded_in :person_nameable, polymorphic: true
+
```

Comment:
if we're redoing names, might want to have a quick talk about the names of these fields -> more proper is things like given name rather than first, family name rather than last.